### PR TITLE
Give better error message for when function arity limit exceeded

### PR DESCRIPTION
**Pull request by jonathanfrawley**
Fixes scala/bug#10737

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/parser/Parsers.scala [hunk 1]
```diff
@@ -2429,7 +2442,7 @@ self =>
             val msg = "Use an implicit parameter instead.\nExample: Instead of `def f[A <% Int](a: A)` use `def f[A](a: A)(implicit ev: A => Int)`."
             if (settings.future)
               deprecationWarning(in.offset, s"View bounds are deprecated. $msg", "2.12.0")
-            contextBoundBuf += atPos(in.skipToken())(makeFunctionTypeTree(List(Ident(pname)), typ()))
+            contextBoundBuf += atPos(in.skipToken())(makeSafeFunctionType(List(Ident(pname)), in.offset, typ()))
```
Comment (jonathanfrawley) [inline src/compiler/scala/tools/nsc/ast/parser/Parsers.scala:2445]: I am not sure if `in.offset` is correct here.

**Comment by SethTisue** [comment]
@df3n5 why closed?

**Comment by hrhino** [comment]
Yes, this was looking good.

**Comment by jonathanfrawley** [comment]
@SethTisue @hrhino Replaced with this: https://github.com/scala/scala/pull/6401
I just wanted to rename the source branch, apologies for the extra noise.

**Pull request closed without merging**